### DISCUSSION TITLE
How to integrate a 3rd party script (Matomo for instance) ?

### DISCUSSION THREAD

**Discussion by hlassiege** [vuepress/core#1233] [Q&A]
Hi,

Is there an easy way to add a snippet of code on each page ? 
I'd like to add Matomo (a kind of Google analytics).

**Comment by Mister-Hope** [comment]
Try to read these plugins source code and understand how vuepress works: https://vuejs.press/plugins/manage.html

**Comment by DaveSanders** [comment]
@hlassiege here's what I had to do to add ZenDesk to my documentation / support site:

Client.ts
```
export default defineClientConfig({
  enhance() {
    if (__VUEPRESS_SSR__) return
    let scriptTag = document.createElement('script')
    scriptTag.async = true
    scriptTag.src = `https://v2.zopim.com/?${ __ZENDESK_ID__ }`
    document.body.appendChild(scriptTag)
  },
})
  ```
  
Then to pass the Zendesk id from my environment to the client.ts, I needed to set a variable in my config.ts. (I'm using dotenv)
  
  ```
  export default defineUserConfig({
  define: {
    __ZENDESK_ID__: process.env.ZENDESK_ID,
  })}
  ```
